addons/binding/org.openhab.binding.ambientweather/src/main/java/org/openhab/binding/ambientweather/internal/handler/AmbientWeatherBridgeHandler.java

```diff
@@ -0,0 +1,238 @@
+    public static final int DEVICES_API_TIMEOUT = 20000;
+
+    // Time to wait after failed key validation
+    public static final long KEY_VALIDATION_DELAY = 60L;
+
+    // Job to validate app and api keys
+    @Nullable
+    private ScheduledFuture<?> validateKeysJob;
+
+    // Application key is granted only by request from developer
+    @Nullable
+    private String applicationKey;
+
+    // API key assigned to user in ambientweather.net dashboard
+    @Nullable
+    private String apiKey;
+
+    // Used Ambient Weather real-time API to retrieve weather data
+    // for weather stations assigned to an API key
+    private AmbientWeatherEventListener listener;
+
+    private final Gson gson;
+
+    private Runnable validateKeysRunnable = new Runnable() {
+        @Override
+        public void run() {
+            logger.debug("Validating application and API keys");
+
+            String response = null;
+            try {
+                // Query weather stations (devices) from Ambient Weather
+                String url = devicesUrl.replace("%APPKEY%", getApplicationKey()).replace("%APIKEY%", getApiKey());
+                logger.debug("Bridge: Querying list of devices from ambient weather service");
+                response = HttpUtil.executeUrl("GET", url, DEVICES_API_TIMEOUT);
+                logger.trace("Bridge: Response = {}", response);
+            } catch (IOException e) {
+                // executeUrl throws IOException when it gets a Not Authorized (401) response
+                logger.debug("Bridge: Got IOException: {}", e.getMessage());
+                updateThingStatus(e.getMessage(), "Invalid API or application key");
+                rescheduleValidateKeysJob();
+                return;
+            } catch (IllegalArgumentException e) {
+                logger.debug("Bridge: Got IllegalArgumentException: {}", e.getMessage());
+                updateThingStatus(e.getMessage(), "Unable to get devices");
+                rescheduleValidateKeysJob();
+                return;
+            }
+            try {
+                // Got a response so the keys are good
+                DevicesJson stations = gson.fromJson(response, DevicesJson.class);
+                logger.debug("Bridge: Application and API keys are valid with {} stations", stations.size());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Connecting to service");
+                // Start up the real-time API listener
+                listener.start(applicationKey, apiKey);
+            } catch (JsonSyntaxException e) {
+                logger.debug("Bridge: Got JsonSyntaxException: {}", e.getMessage());
+                updateThingStatus(e.getMessage(), "Error parsing json response");
+                rescheduleValidateKeysJob();
+                return;
+            }
+        }
+    };
+
+    public AmbientWeatherBridgeHandler(Bridge bridge) {
+        super(bridge);
+        gson = new Gson();
```

Comment:
Can you please assign this straight on the field

Reply:
Done